cmd/agent/app/integrations.go

```diff
@@ -217,6 +218,31 @@ func semverToPEP440(version *semver.Version) string {
 	return pep440
 }
 
+func PEP440ToSemver(pep440 string) (*semver.Version, error) {
+	// Note: this is a simplified version that more closely fits how integrations are versioned.
+	matches := pep440VersionStringRe.FindStringSubmatch(pep440)
+	if matches == nil {
+		return nil, fmt.Errorf("invalid format: %s", pep440)
+	}
+	versionString := matches[1]
+	preReleaseType := matches[2]
+	preReleaseNumber := matches[3]
+	if preReleaseType != "" {
+		switch preReleaseType {
+		case "a":
+			versionString = fmt.Sprintf("%s-alpha", versionString)
+		case "b":
+			versionString = fmt.Sprintf("%s-beta", versionString)
+		default:
+			versionString = fmt.Sprintf("%s-%s", versionString, preReleaseType)
+		}
+		if preReleaseNumber != "" {
```

Comment:
I don't think having a preReleaseType without a preReleaseNumber is PEP440 valid.

Reply:
I had followed the unit tests for `semverToPEP440`, but seems you're right https://www.python.org/dev/peps/pep-0440/#public-version-identifiers

> Pre-release segment: `{a|b|rc}N`

Reply:
… Actually:

> Implicit pre-release number
>
> **Pre releases allow omitting the numeral** in which case it is implicitly assumed to be 0. The normal form for this is to include the 0 explicitly. This allows versions such as 1.2a which is normalized to 1.2a0.

https://www.python.org/dev/peps/pep-0440/#implicit-pre-release-number

So it can be omitted but should be included as 0.

Reply:
We don't ship releases like 1.0.0b but if that's a possibility in the future we should try to be resilient and treat this as 1.0.0-beta.0 I think

Reply:
I'm updating the PR to disallow empty `preReleaseNumber`.